instrumentation/rabbitmq-2.7/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/rabbitmq/DeliveryRequestGetter.java

```diff
@@ -7,21 +7,32 @@
 
 import io.opentelemetry.context.propagation.TextMapGetter;
 import java.util.Collections;
+import java.util.Map;
 
 enum DeliveryRequestGetter implements TextMapGetter<DeliveryRequest> {
   INSTANCE;
 
   @Override
   public Iterable<String> keys(DeliveryRequest carrier) {
-    return carrier != null
-        ? carrier.getProperties().getHeaders().keySet()
-        : Collections.emptyList();
+    if (carrier != null) {
+      Map<String, Object> headers = carrier.getProperties().getHeaders();
+      if (headers == null) {
+        return Collections.emptyList();
+      }
+      return headers.keySet();
+    } else {
+      return Collections.emptyList();
+    }
```

Comment:
WDYT about returning early and reducing the conditional depth a little?

```suggestion
    if (carrier == null) {
      return Collections.emptyList();
    }
    Map<String, Object> headers = carrier.getProperties().getHeaders();
    if (headers == null) {
      return Collections.emptyList();
    }
    return headers.keySet();
```